### PR TITLE
[RHCLOUD-25194] Adding discovery collector to collect the necessary information for indexing

### PR DESCRIPTION
**Pull request by tahmidefaz**
- Parses the necessary data from `Discovery.yml`
- Uses the [Search Platform's CanonicalJson](https://docs.engineering.redhat.com/display/DXPIR/Search+Platform+Canonical+Json+Format+and+Guidelines) format to write the information to `public/canonical.json` file
- This makes it available under the `/api-catalog/canonical.json` endpoint

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "clean": "npm run clean --workspaces --if-present",
     "discovery": "npm-run-all discovery:build discovery:start",
     "discovery:build": "npm run --workspace packages/common --workspace packages/transform build",
-    "discovery:start": "npm run --workspace packages/transform start -- --input-dir `npm prefix`/packages/discovery -o `npm prefix`/packages/common/config",
+    "discovery:start": "npm run --workspace packages/transform start -- --input-dir `npm prefix`/packages/discovery -o `npm prefix`/packages/common/config -p `npm prefix`/public",
```
Comment (josejulio) [inline package.json:18]: Could we move it to the sitemap creation or prerender? That happens at build time

**Review by josejulio — CHANGES_REQUESTED** [review]

File: packages/transform/src/collector.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const APIBASEURL = "https://developers.redhat.com/api-catalog"
+
+type Document = {
+    solr_command: string;
+    content_type: string;
+    id: string;
+    uri: string;
+    name: string;
+    description: string;
```
[the remark this answers is not in the document]
Comment (tahmidefaz) [reply packages/transform/src/collector.ts:11]: Yeah, I think it may be possible to introduce an array of strings here.

File: package.json (hunk 1 above)
Comment (tahmidefaz) [reply package.json:18]: Looking into it :eyeglasses:

**Comment by tahmidefaz** [comment]
@josejulio moved the `collector` under the `sitemap` package and introduces a string array (`content_summaries`) in `canonical.json`

File: public/canonical.json [hunk 3]
```diff
@@ -0,0 +1 @@
...ase Management API","content_summaries":["Get all attachment metadata","Upload a case attachment","Upload request status of all attachments for a given case number","Delete attachment metadata","Check case attachment metadata exist or not","Upload request status","Search cases based on Search Criteria","Search based on Search Criteria","Fetches Recommendation based on Search Criteria","Create Recommendations","Search cases based on Search Criteria","Search based on Search Criteria","Find account by calling User","List Customer Accounts for Partner","List Partner Accounts for Customer","Find account by number","Retrieve list of users for given case group","Find contacts for an account","Get Business Hours","Create new case","Get Cases by filters","Get Case By Case Number","Update case","Get Comments on case","Create Comment on case","Get case comment by comment id","List the externalUpdates on the specified case","Add notified users to case","Delete notified user from case","List Sftp Attachments","Get status for SFTP Attachment uploaded by user","Update token for SFTP User","Reset token for SFTP User","Create upload only SFTP Token for one time use","Create token for SFTP User"]}]}
```
Comment (josejulio) [inline public/canonical.json:1]: It's really good to be able to see this output once, but lets delete and ignore this file. 
Like the sitemap.xml is a build artifact.

Comment (tahmidefaz) [reply public/canonical.json]: done!

**Comment by tahmidefaz** [comment]
@josejulio just removed the `canonical.json` file from under the `public` dir. Ready for review again! :+1:

**Review by josejulio — APPROVED** [review]
Perfecto!

**Pull request merged**